client/cpp/encoder.cc

```diff
@@ -256,4 +331,89 @@ bool Encoder::EncodeString(const std::string& value, Bits* irr_out) const {
   return _EncodeStringInternal(value, &unused_bloom, &unused_prr, irr_out);
 }
 
+static uint8_t shifted(const Bits& bits, const int& index) {
+  // For an array of bytes, select the appopriate byte from a 4-byte
+  // integer value.
+  int result = 0;
+  int shift = 8 * (3 - (index % 4)); // Byte 0 shifts by 24 bits, 1 by 16, etc.
+  result = bits & (0xFF << shift);  // Select the appropriate byte
+  result = result >> shift;          // Shift to LSB
```

Comment:
I'd would have written the last two lines as one: result = (bits >> shift) 0xFF;